src/infer_request.cc

```diff
@@ -442,6 +450,10 @@ InferenceRequest::RespondIfError(
           std::move(response), TRITONSERVER_RESPONSE_COMPLETE_FINAL, status),
       (request->LogRequest() + "failed to send error response").c_str());
 
+  // Report error with type.
+  FailureReason reason = stringToFailureReason(status.Message());  // Convert error message to reason
```

Comment:
I don't think this `status` will contain the expected string for conversion here

Reply:
We can probably pass the `FailureReason` directory to the call to RespondIfError depending on the part of the code we're calling from

Reply:
Look into how many times RespondIfError is called, do some research here.